pkg/api/utils.go

```diff
@@ -19,47 +21,45 @@ const (
 	TenantIDPrefix = "t-"
 	// ContentJSONPatch is the content type to do JSON updates
 	ContentJSONPatch = "application/merge-patch+json"
-
-	idCharset = "abcdefghijklmnopqrstuvwxyz" + "0123456789"
 )
 
 // GenerateClusterID creates a new cluster id
 func GenerateClusterID() (ClusterId, error) {
-	id, err := generateID()
-	if err != nil {
-		return ClusterId{}, err
-	}
+	id, err := generateID(ClusterIDPrefix)
 	return ClusterId{
-		Id: Id(ClusterIDPrefix + id),
-	}, nil
+		Id: id,
+	}, err
 }
 
 // GenerateTenantID creates a new tenant id
 func GenerateTenantID() (TenantId, error) {
-	id, err := generateID()
-	if err != nil {
-		return TenantId{}, err
-	}
+	id, err := generateID(TenantIDPrefix)
 	return TenantId{
-		Id: Id(TenantIDPrefix + id),
-	}, nil
+		Id: id,
+	}, err
 }
 
 // GenerateID generates a new id from random alphanumeric characters
-func generateID() (Id, error) {
-	id := strings.Builder{}
-	for i := 0; i < 6; i++ {
-		max := big.NewInt(int64(len(idCharset)))
-		r, err := rand.Int(rand.Reader, max)
-		if err != nil {
-			return "", err
-		}
-		_, err = id.WriteString(string(idCharset[r.Int64()]))
-		if err != nil {
-			return "", err
+func generateID(prefix string) (Id, error) {
+	min := 10000
+	max := 99999
+	retry := 10
+	var id = ""
+	var i int
+	// let's try a few times
+	for i = 0; i < retry; i++ {
+		h := haikunator.NewHaikunator()
```

Comment:
let's instantiate a global var with it, since it uses the timestamp as rand initialisation: https://github.com/taion809/haikunator/blob/master/haikunator.go#L36